fontbe/src/glyphs.rs

```diff
@@ -278,3 +377,170 @@ fn path_el_type(el: &PathEl) -> &'static str {
+    Rect {
+        x0: bbox.x_min.into(),
+        y0: bbox.y_min.into(),
+        x1: bbox.x_max.into(),
+        y1: bbox.y_max.into(),
+    }
+}
+
+fn rect2bbox(rect: Rect) -> Bbox {
+    Bbox {
+        x_min: rect.min_x() as i16,
+        y_min: rect.min_y() as i16,
+        x_max: rect.max_x() as i16,
+        y_max: rect.max_y() as i16,
+    }
+}
+
+struct GlyphMergeWork {}
+
+pub fn create_glyph_merge_work() -> Box<BeWork> {
+    Box::new(GlyphMergeWork {})
+}
+
+fn compute_composite_bboxes(context: &Context) -> Result<(), Error> {
+    let static_metadata = context.ir.get_static_metadata();
+    let glyph_order = &static_metadata.glyph_order;
+
+    let glyphs: HashMap<_, _> = glyph_order
+        .iter()
+        .map(|gn| (gn, context.get_glyph(gn)))
+        .collect();
+
+    // Simple glyphs have bbox set. Composites don't.
+    // Ultimately composites are made up of simple glyphs, lets figure out the boxes
+    let mut bbox_acquired: HashMap<GlyphName, Bbox> = HashMap::new();
+    let mut composites = glyphs
+        .iter()
+        .filter_map(|(name, glyph)| {
+            let glyph = glyph.as_ref();
+            match glyph {
+                Glyph::Composite(composite) => Some(((*name).clone(), composite.clone())),
+                Glyph::Simple(..) => None,
+            }
+        })
+        .collect::<Vec<_>>();
+
+    trace!("Resolve bbox for {} composites", composites.len());
+    while !composites.is_empty() {
+        let pending = composites.len();
+
+        // Hopefully we can figure out some of those bboxes!
+        for (glyph_name, composite) in composites.iter() {
+            let mut missing_boxes = false;
+            let boxes: Vec<Bbox> = composite
+                .components()
+                .iter()
+                .filter_map(|c| {
+                    if missing_boxes {
+                        return None; // can't succeed
+                    }
+                    let ref_glyph_name = glyph_order.get_index(c.glyph.to_u16() as usize).unwrap();
+                    let bbox = bbox_acquired.get(ref_glyph_name).copied().or_else(|| {
+                        glyphs
+                            .get(ref_glyph_name)
+                            .map(|g| g.as_ref().clone())
+                            .and_then(|g| match g {
+                                Glyph::Composite(..) => None,
+                                Glyph::Simple(simple_glyph) => Some(simple_glyph.bbox),
+                            })
+                    });
+                    if bbox.is_none() {
+                        trace!("Can't compute bbox for {glyph_name} because bbox for {ref_glyph_name} isn't ready yet");
+                        missing_boxes = true;
+                        return None; // maybe next time?
+                    };
+
+                    // The transform we get here has changed because it got turned into F2Dot14 and i16 parts
+                    // We could go get the "real" transform from IR...?
```

Comment:
I think we should...

Reply:
I think I'd like to make that a followup PR, filed as https://github.com/googlefonts/fontmake-rs/issues/135